packages/aws-cdk-lib/aws-cloudfront/lib/response-headers-policy.ts

```diff
@@ -114,13 +114,36 @@ export class ResponseHeadersPolicy extends Resource implements IResponseHeadersP
       maxLength: 128,
     });
 
+    let securityHeadersBehavior = props.securityHeadersBehavior;
+    let customHeadersBehavior = props.customHeadersBehavior;
+
+    if (securityHeadersBehavior?.contentSecurityPolicy?.reportOnly) {
+      const reportOnlyCSPHeader = {
+        header: 'Content-Security-Policy-Report-Only',
+        value: securityHeadersBehavior.contentSecurityPolicy.contentSecurityPolicy,
+        override: true,
+      };
+      securityHeadersBehavior = {
+        ...securityHeadersBehavior,
+        contentSecurityPolicy: undefined,
+      };
+
+      if (!customHeadersBehavior) {
+        customHeadersBehavior = {
+          customHeaders: [],
+        };
```

Comment:
Why are you making this an empty array only if `reportOnly` is set?

Reply:
The `Content-Security-Policy-Report-Only` header is only respected as a custom header configuration. The following line is pushing the report only csp header into this `customHeaders` array. If the `customHeadersBehavior` object is undefined (which it's allowed to be), we wont have anything to push into.
In the non-report-only configuration case, we don't need to worry about touching the `customHeaders` array.